HW06-Annotation/src/main/java/com/akproductions/testframework/TestFramework.java

```diff
@@ -0,0 +1,85 @@
+package com.akproductions.testframework;
+
+import com.akproductions.testframework.annotations.After;
+import com.akproductions.testframework.annotations.Before;
+import com.akproductions.testframework.annotations.Test;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class TestFramework implements TestResult{
+    private Class clazz;
+    private TestResults testResults;
+    public TestFramework (String classToTest) throws Exception {
+        clazz = Class.forName(classToTest);
+//        testResults= new HashMap<String,Integer> ();
+        testResults= new TestResults <String,Integer> ();
+    }
+
+    public Map<String, Integer> doTest () throws Exception
+    {
+        Method[] methodsAll = clazz.getDeclaredMethods();
+        var methodsBefore = new ArrayList<Method>(10);
+        var methodsAfter = new ArrayList<Method>(10);
+        var methodsTest = new ArrayList<Method>(10);
+        for (Method method : methodsAll) {
+            Annotation[] annotations = method.getDeclaredAnnotations();
+            for (Annotation annotation : annotations) {
+                if (annotation.annotationType().equals(Before.class)) methodsBefore.add(method);
+                if (annotation.annotationType().equals(After.class)) methodsAfter.add(method);
+                if (annotation.annotationType().equals(Test.class)) methodsTest.add(method);
+            }
+            }
+        System.out.println("[INFO] --- Before methods:");
+        methodsBefore.forEach(c ->   System.out.println(c.getName()));
+        System.out.println("[INFO] --- After methods:");
+        methodsAfter.forEach(c ->   System.out.println(c.getName()));
+        System.out.println("[INFO] --- Test methods:");
+        methodsTest.forEach(c ->   System.out.println(c.getName()));
+
+        return makeTests(methodsTest, methodsBefore, methodsAfter);
+    }
+
+    private Map<String, Integer> makeTests(ArrayList<Method> methodsTest,ArrayList<Method> methodsBefore, ArrayList<Method> methodsAfter) throws Exception
```

Comment:
```suggestion
    private Map<String, Integer> makeTests(List<Method> methodsTest, List<Method> methodsBefore, List<Method> methodsAfter) throws Exception
```

Интерфейсы вместо реализаций